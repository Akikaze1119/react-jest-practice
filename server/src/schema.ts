import { gql } from 'graphql-tag';
import { nanoid } from 'nanoid';

export const typeDefs = gql`
  type Todo {
    id: ID!
    title: String!
    completed: Boolean!
  }

  type Query {
    todos: [Todo!]!
  }

  type Mutation {
    addTodo(title: String!): Todo!
    toggleTodo(id: ID!): Todo!
    removeTodo(id: ID!): ID!
  }
`;

let todos = [
  { id: '1', title: 'First Todo', completed: false },
  { id: '2', title: 'Second Todo', completed: true },
  { id: '3', title: 'Third Todo', completed: false },
];

export const resolvers = {
  Query: {
    todos: () => todos,
  },
  Mutation: {
    addTodo: (_: any, { title }: { title: string }) => {
      const newTodo = { id: nanoid(), title, completed: false };
      todos.push(newTodo);
      return newTodo;
    },

    toggleTodo: (_: any, { id }: { id: string }) => {
      const todo = todos.find((t) => t.id === id);
      if (todo) todo.completed = !todo.completed;
      return todo;
    },

    removeTodo: (_: any, { id }: { id: string }) => {
      todos = todos.filter((t) => t.id !== id);
      return id;
    },
  },
};
