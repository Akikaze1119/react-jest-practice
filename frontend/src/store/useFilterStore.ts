import { create } from 'zustand';

type TFilterState = {
  filter: 'all' | 'completed' | 'incomplete';
  setFilter: (filter: 'all' | 'completed' | 'incomplete') => void;
};

export const useFilterStore = create<TFilterState>((set) => ({
  filter: 'all',
  setFilter: (filter) => set({ filter }),
}));
