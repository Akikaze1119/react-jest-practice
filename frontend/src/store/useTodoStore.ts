import { create } from 'zustand';
import { TTodo } from '../types/Todo';

type TTodoStoreState = {
  todos: TTodo[];
  addTodo: (title: string) => void;
  toggleTodo: (id: number) => void;
  removeTodo: (id: number) => void;
};

export const useTodoStore = create<TTodoStoreState>((set) => ({
  todos: [
    { id: 1, title: 'Learn React', completed: false },
    { id: 2, title: 'Learn Zustand', completed: false },
    { id: 3, title: 'Build a Todo App', completed: true },
  ],

  addTodo: (title: string) =>
    set((state) => ({
      todos: [...state.todos, { id: Date.now(), title, completed: false }],
    })),

  toggleTodo: (id: number) =>
    set((state) => ({
      todos: state.todos.map((todo) =>
        todo.id === id ? { ...todo, completed: !todo.completed } : todo
      ),
    })),

  removeTodo: (id: number) =>
    set((state) => ({
      todos: state.todos.filter((todo) => todo.id !== id),
    })),
}));
