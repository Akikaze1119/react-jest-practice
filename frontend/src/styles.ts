import { createGlobalStyle } from 'styled-components';
import { redux } from 'zustand/middleware';

export const colors = {
  primary: '#0ae47b',
  red: '#c32c3e',
  black: '#131313',
  gray: '#cecece',
};

export const GlobalStyle = createGlobalStyle`
  * {
    box-sizing: border-box;
  }

  body {
    background: #000;
    color: #fff;
    font-family: Poppins, sans-serif;
  }

  a {
    color: inherit;
    text-decoration: none;
  }

  button {
    cursor: pointer;
    font-family: inherit;
    font-size: 1rem;
    border: none;
  }

  input {
    font-family: inherit;
    font-size: 1rem;
  }

  input:focus {
    outline: none;
  }

  ul {
    list-style: none;
    padding: 0;
    margin: 0;
  }

  li {
    display: flex;
    align-items: center;
    padding: 0.5rem 0;
  }

  h1 {
    font-size: 2rem;
    font-weight: 400;
    margin: 0;
    padding: 1rem;
    text-align: center;
  }
`;
