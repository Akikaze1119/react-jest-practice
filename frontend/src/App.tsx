import { styled } from 'styled-components';

import { TodoList } from './components/TodoList';
import { TodoAdd } from './components/TodoAdd';
import { FilterButtons } from './components/FilterButtons';

import { GlobalStyle } from './styles';

const Layout = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 2rem;
  max-width: 800px;
  margin: 0 auto;
`;

export function App() {
  return (
    <Layout>
      <GlobalStyle />
      <h1>Todo List</h1>
      <FilterButtons />
      <TodoList />
      <TodoAdd />
    </Layout>
  );
}
