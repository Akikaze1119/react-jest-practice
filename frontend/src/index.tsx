import { createRoot } from 'react-dom/client';
import { StrictMode } from 'react';

import { ApolloProvider } from '@apollo/client';
import { client } from './apolloClient'; // Assuming you have an Apollo Client setup

import './index.css';
import { App } from './App';

let container = document.getElementById('app')!;
let root = createRoot(container);
root.render(
  <StrictMode>
    <ApolloProvider client={client}>
      <App />
    </ApolloProvider>
  </StrictMode>
);
