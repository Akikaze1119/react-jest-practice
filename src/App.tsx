import './App.css';
import { Counter } from './components/Counter';
import { StyledButton } from './components/StyledButton';
import { Greeting } from './components/Greeting';

const testButton = () => {
  console.log('Button clicked!');
};

export function App() {
  return (
    <>
      <h1>Hello from App component!</h1>
      <h2>Greeting Component</h2>
      <Greeting name='John Doe' />

      <h2>Styled-components</h2>
      <StyledButton handleClick={testButton} label='button' />

      <h2>Test Zustand</h2>
      <Counter />
    </>
  );
}
