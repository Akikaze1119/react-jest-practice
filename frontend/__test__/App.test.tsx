import { render } from '@testing-library/react';
import { App } from '../src/App';
import { useTodoStore } from '../src/store/useTodoStore';

describe('<App/>', () => {
  it('renders correctly (snapshot)', () => {
    useTodoStore.setState({
      todos: [
        { id: 1, title: 'Test Todo', completed: false },
        { id: 2, title: 'Another Todo', completed: true },
      ],
    });

    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
});
