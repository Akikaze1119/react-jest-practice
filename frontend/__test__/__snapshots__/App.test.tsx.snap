// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<App/> renders correctly (snapshot) 1`] = `
<div>
  <div>
    <h1>
      Todo List
    </h1>
    <form>
      <input
        placeholder="Add a new todo"
        value=""
      />
      <button
        type="submit"
      >
        Add Todo
      </button>
    </form>
    <ul>
      <li
        class="sc-bRKDuR kPCVXD"
      >
        Test Todo
        <button>
          Delete
        </button>
      </li>
      <li
        class="sc-bRKDuR ibaPth"
      >
        Another Todo
        <button>
          Delete
        </button>
      </li>
    </ul>
  </div>
</div>
`;
